src/components/App/App.js

```diff
@@ -0,0 +1,54 @@
+import classnames from 'classnames'
+import React, { useState, useEffect } from 'react'
+import { useHistory } from 'react-router-dom'
+import AppRouter from '../AppRouter/AppRouter'
+import Sidebar from '../Sidebar/Sidebar'
+import { SidebarProvider } from '../Sidebar/SidebarContext'
+import styles from './App.module.css'
+
+const useSidebar = () => {
+    const [visible, setVisible] = useState(true)
+    return {
+        visible,
+        show: () => {
+            setVisible(true)
+        },
+        hide: () => {
+            setVisible(false)
+        },
+    }
+}
```

Comment:
Maybe we just prefer different styles, but personally I always try have a single component per file, with hooks and subcomponents in separate files. I personally find it easier to read that way. Let me know what you think, because that might just be me who finds that easier to process.

Reply:
I've moved the hook to a separate file :+1: I tend to move extract hooks and subcomponents only when I feel they reach a certain level of complexity, but since we want consistent code architecture across our apps it makes sense to follow a more consistent rule such as one component per file.